content/client-lib-development-guide/features.textile

```diff
@@ -1242,6 +1245,20 @@ h4. TokenRequest
 * @(TE4)@ @ttl@ attribute represents time to live (expiry) of this token in milliseconds
 * @(TE6)@ @TokenRequest::fromJson@ static factory method that accepts either a @JsonObject@ or a string (which should be parsed as a JSON string), and returns a new @TokenRequest@. Statically typed languages (that expect the @authCallback@ to result in a typed @TokenRequest@ object, rather than, say, a hashmap) must implement this; others may at their discretion.
 
+h4. TokenRevocationRequest
+
+* @(TF1)@ @TokenRevocationRequest@ is a type containing the token revocation request details sent to the "REST revokeTokens endpoint":/rest-api/#revokeTokens
+* @(TF2)@ @TokenRevocationRequest#keys@ attribute is an array of strings containing the keys to revoke. For request to be valid, it must contain at least one key. A test must exist to ensure there is at least one key in the request before it is sent.
```

Comment:
I think we should generally avoid adding statements that say "A test must exist ....".

The REST endpoint is `POST /keys/appId.keyId/revokeTokens` which revokes tokens associated with a specific key, whereas this request permits multiple keys. This introduces complexity because there is the possibility that the caller has rights to revoke tokens for one key but not others. I suggest we restrict this API to mirror the REST API, and only support token revocation for a single key.

Reply:
Thank you. I think reading spec and following and example I mistook the . with , and somehow  thought we support multiple keys. I made some changes here 1c220a6fc382fa4744baec3a9e5796e16013b431

Reply:
@paddybyers If this conversation needs to continue, then that will need to happen against the new pull request that we will open against https://github.com/ably/specification/issues/10. Please resolve this conversation _here_ once you have re-raised your concern _there_, or if you are already happy that this matter is resolved. Thanks.